azext_iot/monitor/handlers/_internal.py

```diff
@@ -0,0 +1,66 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+import re
+import yaml
+
+from azext_iot.monitor.handlers._parser import Event3Parser
+
+
+def parse_message(
+    msg,
+    device_id,
+    devices,
+    pnp_context,
+    interface_name,
+    content_type,
+    properties,
+    output,
+    validate_messages,
+    simulate_errors,
+    central_device_provider,
+):
+    parser = Event3Parser()
+    origin_device_id = parser.parse_device_id(msg)
+
+    if not _should_process_device(origin_device_id, device_id, devices):
+        return
+
+    parsed_msg = parser.parse_message(
+        msg,
+        pnp_context,
+        interface_name,
+        properties,
+        content_type,
+        simulate_errors,
+        central_device_provider,
```

Comment:
Ultimately the provider for device templates needs to be abstracted and not assume Central (there are services other than Central that deal with device templates).

Reply:
Agreed, I think that'll be the next item I'll address